### PR TITLE
commandURLCleanup now strips all unwanted chars at the end of the string

### PR DESCRIPTION
**Pull request by pkamps**
This came up in the CSM project: ticket 3479.

If you have an object name like "about." (including the double quotes), ezp was not removing the dot after 'about'.

### CODE REVIEW

File: lib/ezi18n/classes/ezchartransform.php [hunk 1]
```diff
@@ -395,14 +395,12 @@ static function commandUrlCleanup( $text, $charsetName )
         $sep  = eZCharTransform::wordSeparator();
         $sepQ = preg_quote( $sep );
         $text = preg_replace( array( "#[^a-zA-Z0-9_!\.-]+#",
-                                     "#^[\.]+|[!\.]+$#", # Remove dots at beginning/end
                                      "#\.\.+#", # Remove double dots
                                      "#[{$sepQ}]+#", # Turn multiple separators into one
-                                     "#^[{$sepQ}]+|[{$sepQ}]+$#" ), # Strip separator from beginning/end
+                                     "#^[{$sepQ}\.]+|[{$sepQ}!\.]+$#" ), # Strip unwanted chars from beginning/end
```
Comment (peterkeung) [inline lib/ezi18n/classes/ezchartransform.php]: This comment needs to be be more specific. It should say "Strip separator and periods from beginning / end"

File: tests/tests/lib/ezi18n/ezchartransform_test.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * File containing the eZCharTransFormTests class
+ *
+ * @package tests
+ */
+
+class eZCharTransFormTests extends ezpTestCase
```
Comment (peterkeung) [inline tests/tests/lib/ezi18n/ezchartransform_test.php]: the "f" should not be capitalized

File: tests/tests/lib/ezi18n/ezchartransform_test.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->setName( "eZCharTransFormTests" );
```
Comment (peterkeung) [inline tests/tests/lib/ezi18n/ezchartransform_test.php]: the "f" should not be capitalized

File: tests/tests/lib/ezi18n/suite.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * @package tests
+ */
+
+class eZFileTestSuite extends ezpTestSuite
+{
+    public function __construct()
+    {
+        parent::__construct();
+        $this->setName( "eZFile Test Suite" );
+        $this->addTestSuite( 'eZCharTransFormTests' );
```
Comment (peterkeung) [inline tests/tests/lib/ezi18n/suite.php]: the "f" should not be capitalized

File: tests/tests/lib/ezi18n/ezchartransform_test.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * File containing the eZCharTransFormTests class
```
Comment (peterkeung) [inline tests/tests/lib/ezi18n/ezchartransform_test.php]: the "f" should not be capitalized

**Comment by pkamps** [comment]
Feedback implemented

File: lib/ezi18n/classes/ezchartransform.php [hunk 6]
```diff
@@ -397,7 +397,7 @@ static function commandUrlCleanup( $text, $charsetName )
         $text = preg_replace( array( "#[^a-zA-Z0-9_!\.-]+#",
                                      "#\.\.+#", # Remove double dots
                                      "#[{$sepQ}]+#", # Turn multiple separators into one
-                                     "#^[{$sepQ}\.]+|[{$sepQ}!\.]+$#" ), # Strip unwanted chars from beginning/end
+                                     "#^[{$sepQ}\.]+|[{$sepQ}!\.]+$#" ), # Strip separator and dot from beginning, strip exclemation mark, dot and separator from end
```
Comment (peterkeung) [inline lib/ezi18n/classes/ezchartransform.php]: typo: should be "exclamation"

**Comment by pkamps** [comment]
Tyop fixed

**Comment by peterkeung** [comment]
+1

**Comment by pkamps** [comment]
So I added the fix also to commandUrlFixIRI. It's also covered by dedicated tests.

For commandUrlFixIRI I also drop the 'double quote' char. That was not in the original patch from eZ but I still think it's a good idea to remove it from the URL.

**Comment by peterkeung** [comment]
This was fixed in the main eZ repo: https://github.com/ezsystems/ezpublish-legacy/commit/6e625bee207f7e1f7a045f954fc18b2559a37c75

+1 with the double quote strip as well. We should create an additional pull request against the eZ repo with the double quote strip.

**Comment by pkamps** [comment]
ez systems pull request:
https://github.com/ezsystems/ezpublish-legacy/pull/1274

**Pull request merged**